### DISCUSSION TITLE
XC. Python 2 support to be discontinued: is this a problem for you?

### DISCUSSION THREAD

**Discussion by lcpt** [xcfem/xc#78] [General]
Well, we are encountering some difficulties to use [type hints](https://www.python.org/dev/peps/pep-0484/) in XC Python modules when using Python 2. It's not impossible [to use that in Python 2](https://stackoverflow.com/questions/35230635/type-hinting-in-python-2), but apparently (I've discovered type hints a few days ago, it's @ebrahimraeyat who has started to use them) this solution is not as complete as the true type hint in Python 3 (see [this comment](https://github.com/xcfem/xc/pull/77#issuecomment-760714080)).

I think that we can use type hints to make the code easier to understand (if we want to make XC attractive, this is important), so maybe the simpler solution is to stop supporting Python 2 from now on. Anyway Python 2 has been discontinued as of January 1, 2020. We can make a last "Python 2 compatible" release if needed.

There is maybe some of you that are still using Python 2 and cannot switch to version 3 for some reason. Please let us know if this is the case.

**Comment by ebrahimraeyat** [comment]
Hi @lcpt. It seems perhaps no one has a problem with this.

**Comment by lcpt** [comment]
Hi @ebrahimraeyat.

Yes, I think so. I'll make a Python 2 release and then merge the PR this weekend.
